lib/pbench/test/unit/server/test_datasets_metadata.py

```diff
@@ -209,6 +209,50 @@ def test_get_bad_query_2(self, query_get_as):
         )
         assert response.json == {"message": "Unknown URL query keys: controller,plugh"}
 
+    def test_get_funky_metalog_key(self, query_get_as):
+        """Test funky metadata.log key
+
+        Normally we constrain metadata keys to lowercase alphanumeric strings.
+        Traditional Pbench Agent `metadata.log` files contain keys constructed
+        from benchmark iteration values that can contain mixed case and symbol
+        characters. We allow these keys to be filtered and retrieved, but not
+        created, so test that we can filter on a funky key value and return
+        the key.
```

Comment:
So, I see tests for the successful use of weird keys in filtering and retrieval, but I don't see a test which proves that we prohibit them from being created/set...should there be one?